### PR TITLE
Add tutorial to explain naming of PyGMT figure elements

### LINKED ISSUE #2026: Explain the naming of figure elements in a PyGMT figure
The figure below explains the anatomy of a matplotlib figure.
image

As you can see, the naming of figure elements is different between matplotlib and PyGMT. 

1. _tick labels_ in matplotlib are called _annotations_ in PyGMT
2. _axes_ in matplotlib is a region for plotting data. I don't think there is an equivalent in PyGMT
3. _axis_ (e.g., x-axis and y-axis)in matplotlib is called _axes_ in PyGMT, and is specified by **WSENwsen**.
4. _spines_ in matplotlib are lines connecting the axis tick marks and noting the boundaries of the data area. In GMT/PyGMT, they're plotted by specifying **l**, **r**, **t**, or **b**.

These differences may confuse users coming from matplotlib. I think it would be good if we can create a similar figure and short tutorial, explaining the naming of figure elements in GMT/PyGMT.

### PR DESCRIPTION
**Pull request by JingHuiTong**
**Description of proposed changes**
Add a tutorial to explain naming of figure elements in a PyGMT
image

**Preview**: https://pygmt-dev--2383.org.readthedocs.build/en/2383/get_started/03_figure_element.html

Fixes #2026

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: examples/tutorials/basics/fig_element.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+"""
```
Comment (seisman) [inline examples/tutorials/basics/fig_element.py:1]: I feel this tutorial should be put into the `examples/get_started` directory.

File: examples/get_started/03_figure_element.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+"""
+Figure elements
+=================================
+The figure shows the naming of figure elements in pygmt.
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:4]:
```suggestion
The figure shows the naming of figure elements in PyGMT. 
```

File: examples/get_started/03_figure_element.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+"""
+Figure elements
+=================================
+The figure shows the naming of figure elements in pygmt. 
+
+* ``frame``: setting map boundary(**WSNE** or **wsne**), annonate and grid(**afg**), axis label(**+l**) and title(**+t**) in :func:`pygmt.Figure.basemap`
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:6]:
```suggestion
* ``frame``: setting map boundary (**WSNE** or **wsne**), annonate and grid (**afg**), axis label (**+l**) and title (**+t**) in :func:`pygmt.Figure.basemap`
```

File: examples/get_started/03_figure_element.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
+"""
+Figure elements
+=================================
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:3]:
```suggestion
Figure elements
===============
```

File: examples/get_started/03_figure_element.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+The figure shows the naming of figure elements in pygmt. 
+
+* ``frame``: setting map boundary(**WSNE** or **wsne**), annonate and grid(**afg**), axis label(**+l**) and title(**+t**) in :func:`pygmt.Figure.basemap`
+* :func:`pygmt.Figure.plot`: plotting the lines or symbols based on ``pen`` or ``style`` parameters, respectively
+* :func:`pygmt.Figure.legend`: showing the naming of lines or symbols while the ``label`` given in :func:`pygmt.Figure.plot`
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:8]:
```suggestion
* ``frame``: setting map boundary (**WSNE** or **wsne**), annonate and grid (**afg**), axis label (**+l**) and title (**+t**) in :meth:`pygmt.Figure.basemap`
* :meth:`pygmt.Figure.plot`: plotting the lines or symbols based on ``pen`` or ``style`` parameters, respectively
* :meth:`pygmt.Figure.legend`: showing the naming of lines or symbols while the ``label`` given in :meth:`pygmt.Figure.plot`
```

File: examples/get_started/03_figure_element.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+# ============fig.plot
+fig.plot(x=6, y=8, style="c0.7c", pen="2p,darkblue")
+fig.plot(x=4, y=6, style="c0.7c", pen="2p,darkblue")
+fig.text(x=5, y=6, text="fig.plot()", font="10p,8", justify="BL")
+
+
+fig.legend()
+# ============Legned
+fig.text(x=7.5, y=17, text="Legend", font="12p,2,darkblue", justify="TL", no_clip=True)
+fig.text(x=7.5, y=15.8, text="fig.legend()", font="10p,8", justify="TL", no_clip=True)
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:85]:
```suggestion

# ============Legend
fig.legend()
fig.text(x=7.5, y=17, text="Legend", font="12p,2,darkblue", justify="TL", no_clip=True)
fig.text(x=7.5, y=15.8, text="fig.legend()", font="10p,8", justify="TL", no_clip=True)
```

**Comment by willschlitzer** [comment]
@JingHuiTong Do you get any changes when you run `make format` and `make check`? That should help you with the failing Style Checks.

File: examples/get_started/03_figure_element.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  (**afg**), axis label (**+l**) and title (**+t**) in
+  :meth:`pygmt.Figure.basemap`. The detail examples can be found at
+  `frame and axes attributes<https://www.pygmt.org/latest/tutorials/basics/
+  frames.html>`_.
+- :meth:`pygmt.Figure.plot`: plotting the lines or symbols based on ``pen``
+  or ``style`` parameters, respectively
+- :meth:`pygmt.Figure.text`: plottong text strings with ``font`` parameter
+  which adjusts the fontsize, fontstyle and color.
+- :meth:`pygmt.Figure.legend`: showing the naming of lines or symbols while
+  the ``label`` given in :meth:`pygmt.Figure.plot`
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:18]:
```suggestion
- :meth:`pygmt.Figure.legend`: showing the naming of lines or symbols while
  the ``label`` is given in :meth:`pygmt.Figure.plot`
```

File: examples/get_started/03_figure_element.py [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  must start with the creation of a :meth:`pygmt.Figure` instance
+- ``frame``: setting map boundary (**WSNE** or **wsne**), annonate and grid
+  (**afg**), axis label (**+l**) and title (**+t**) in
+  :meth:`pygmt.Figure.basemap`. The detail examples can be found at
+  `frame and axes attributes<https://www.pygmt.org/latest/tutorials/basics/
+  frames.html>`_.
+- :meth:`pygmt.Figure.plot`: plotting the lines or symbols based on ``pen``
+  or ``style`` parameters, respectively
+- :meth:`pygmt.Figure.text`: plottong text strings with ``font`` parameter
+  which adjusts the fontsize, fontstyle and color.
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:16]:
```suggestion
- :meth:`pygmt.Figure.text`: plotting text strings whereby the ``font``
  parameter adjusts fontsize, fontstyle, and color.
```

File: examples/get_started/03_figure_element.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
+"""
+3. Figure elements
+===============
+The figure shows the naming of figure elements in PyGMT.
+
+- :meth:`pygmt.Figure`: having a number of plotting methods. Every plot or map
+  must start with the creation of a :meth:`pygmt.Figure` instance
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:7]: Users can add different plots or / and maps to one figure object.
```suggestion
- :meth:`pygmt.Figure`: having a number of plotting methods. Every new figure
  must start with the creation of a :meth:`pygmt.Figure()` instance
```

File: examples/get_started/03_figure_element.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+- ``frame``: setting map boundary (**WSNE** or **wsne**), annonate and grid
+  (**afg**), axis label (**+l**) and title (**+t**) in
+  :meth:`pygmt.Figure.basemap`. The detail examples can be found at
+  `frame and axes attributes<https://www.pygmt.org/latest/tutorials/basics/
+  frames.html>`_.
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:12]: Beside `pygmt.Figure.basemap` also other methods have the ``frame`` parameter.
```suggestion
- `
- ``frame``: setting plot boundaries (**WSNE**, **wsne**, or **lbtr**),
   adding annotations, ticks, gridlines (**afg**), axis labels (**+l**),
   and title (**+t**), e.g., in :meth:`pygmt.Figure.basemap`. Detailed
   examples can be found at
   `frame and axes attributes<https://www.pygmt.org/latest/tutorials/basics/frames.html>`_.
```

File: examples/get_started/03_figure_element.py [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+fig.text(x=5, y=-3, text='frame="x+lxlabel"', **minorexplain)
+# ============ylabel
+fig.text(x=-1.7, y=10, text='frame="y+lylabel"', angle=90, **minorexplain)
+# ============x-majorticks
+fig.plot(x=10, y=-0.2, style="c1c", pen="2p,darkblue", no_clip=True)
+fig.text(x=10, y=-1.4, text="Annonate", **mainexplain)
+fig.text(x=10, y=-2.6, text='frame="xa2"', **minorexplain)
+# ============y-majorticks
+fig.plot(x=-0.2, y=20, style="c1c", pen="2p,darkblue", no_clip=True)
+fig.text(x=0, y=23.5, text="Annonate", **mainexplain)
```
Comment (weiji14) [inline examples/get_started/03_figure_element.py:54]: Typo. Should it be Annotate or Annotation?
```suggestion
fig.text(x=10, y=-1.4, text="Annotation", **mainexplain)
fig.text(x=10, y=-2.6, text='frame="xa2"', **minorexplain)
# ============y-majorticks
fig.plot(x=-0.2, y=20, style="c1c", pen="2p,darkblue", no_clip=True)
fig.text(x=0, y=23.5, text="Annotation", **mainexplain)
```

File: examples/get_started/03_figure_element.py [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+==================
+The figure shows the naming of figure elements in PyGMT.
+
+- :meth:`pygmt.Figure()`: having a number of plotting methods. Every new
+  figure must start with the creation of a :meth:`pygmt.Figure()` instance
+- ``frame``: setting plot or map boundaries (**WSNE**, **wsne**, or
+  **lbtr**), adding annotations, ticks, gridlines (**afg**), axis labels
+  (**+l**), and title (**+t**), e.g., in :meth:`pygmt.Figure.basemap`.
+  Detailed examples can be found at `frame and axes attributes
+  <https://www.pygmt.org/latest/tutorials/basics/frames.html>`_.
```
Comment (weiji14) [inline examples/get_started/03_figure_element.py:12]: Use relative intersphinx link - https://docs.readthedocs.io/en/stable/guides/intersphinx.html#using-intersphinx
```suggestion
  Detailed examples can be found at :doc:`frame and axes attributes
  </tutorials/basics/frames>`.
```

File: examples/get_started/03_figure_element.py [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+fig.text(x=2, y=17, text="Grid", **mainexplain)
+fig.text(x=2, y=18, text='frame="xg2"', **minorexplain)
+# ============map boundaries
+fig.plot(x=10, y=9, style="c0.5c", pen="2p,darkblue", no_clip=True)
+fig.text(x=11.5, y=8, text="Map Boundary", **mainexplain)
+fig.text(x=11.5, y=7, text='frame="WSne"', **minorexplain)
+# ============fig.plot
+fig.plot(x=6, y=8, style="c0.7c", pen="2p,darkblue")
+fig.plot(x=4, y=6, style="c0.7c", pen="2p,darkblue")
+fig.text(x=6, y=6.5, text="fig.plot()", **minorexplain)
```
Comment (weiji14) [inline examples/get_started/03_figure_element.py:74]: Explain the triangle plot (style) and red line (pen) separately?

```suggestion
# ============fig.plot (style)
fig.plot(x=6, y=8, style="c0.7c", pen="2p,darkblue")
fig.text(x=6, y=6.5, text="fig.plot(style)", **minorexplain)
# ============fig.plot (pen)
fig.plot(x=4, y=6, style="c0.7c", pen="2p,darkblue")
fig.text(x=3, y=4.5, text="fig.plot(pen)", **minorexplain)
```

**Review by weiji14 — COMMENTED** [review]
Hi @JingHuiTong, sorry for the slow response. This example is looking pretty good already, and we're hoping to maybe get it in for the PyGMT v0.10.0 release due next week (1 Sep) if possible. Was there anything else you wanted to add to the example, or can we start to finalize things?

File: examples/get_started/03_figure_element.py [hunk 14]
```diff
@@ -0,0 +1,80 @@
+"""
+3. Figure elements
+==================
+The figure shows the naming of figure elements in PyGMT.
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:4]:
```suggestion
The figure below shows the naming of figure elements in PyGMT.
```

File: examples/get_started/03_figure_element.py [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+
+- :meth:`pygmt.Figure()`: having a number of plotting methods. Every new
+  figure must start with the creation of a :meth:`pygmt.Figure()` instance
+- ``frame``: setting plot or map boundaries (**WSNE**, **wsne**, or
+  **lbtr**), adding annotations, ticks, gridlines (**afg**), axis labels
+  (**+l**), and title (**+t**), e.g., in :meth:`pygmt.Figure.basemap`.
+  Detailed examples can be found at `frame and axes attributes
+  <https://www.pygmt.org/latest/tutorials/basics/frames.html>`_.
+- :meth:`pygmt.Figure.plot`: plotting the lines or symbols based on ``pen``
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:13]:
```suggestion
- :meth:`pygmt.Figure.plot`: plotting lines or symbols based on ``pen``
```

File: examples/get_started/03_figure_element.py [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+  or ``style`` parameters, respectively
+- :meth:`pygmt.Figure.text`: plotting text strings whereby the ``font``
+  parameter adjusts fontsize, fontstyle, and color.
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:16]:
```suggestion
  parameter adjusts fontsize, fontstyle, and color
```
To be consistent with the the other points.

File: examples/get_started/03_figure_element.py [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+- :meth:`pygmt.Figure.legend`: showing the naming of lines or symbols while
+  the ``label`` is given in :meth:`pygmt.Figure.plot`
+- :meth:`pygmt.Figure.show`: previewing the figure you plotted
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:19]:
```suggestion
- :meth:`pygmt.Figure.show`: previewing the content added to the current figure instance
```

File: examples/get_started/03_figure_element.py [hunk 18]
```diff
@@ -0,0 +1,80 @@
+"""
+3. Figure elements
+==================
+The figure shows the naming of figure elements in PyGMT.
+
+- :meth:`pygmt.Figure()`: having a number of plotting methods. Every new
+  figure must start with the creation of a :meth:`pygmt.Figure()` instance
+- ``frame``: setting plot or map boundaries (**WSNE**, **wsne**, or
+  **lbtr**), adding annotations, ticks, gridlines (**afg**), axis labels
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:9]: Hm, I am not completely happy with the formulation of this bullet point. Actually, the user can pass a combination of the single letters of `WSENE`, `wsne`, and `lbtr`.

**Comment by JingHuiTong** [comment]
> 

Sorry for the slow response, I think the example almost done. I will try to review the comment.

File: examples/get_started/03_figure_element.py [hunk 19]
```diff
@@ -0,0 +1,83 @@
+"""
+3. Figure elements
+==================
+The figure below shows the naming of figure elements in PyGMT.
+
+- :meth:`pygmt.Figure()`: having a number of plotting methods. Every new
+  figure must start with the creation of a :meth:`pygmt.Figure()` instance
+- ``frame``: setting plot or map boundaries (combination of the single 
+  letter of **WSNE**, **wsne**, or **lbtr**), adding annotations, ticks,
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:9]:
```suggestion
- ``frame``: setting plot or map boundaries (a combination of the single 
  letters of **WSNE**, **wsne**, or **lbtr**), adding annotations, ticks, 
```

**Review by weiji14 — APPROVED** [review]
Thanks @JingHuiTong! We'll merge this after a second approval, and get it in for PyGMT v0.10.0 :smile:

File: examples/get_started/03_figure_element.py [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+fig.text(x=11.5, y=6.5, text='frame="WStr"', **minorexplain)
+# ============fig.plot (style)
+fig.plot(x=6, y=8, style="c0.7c", pen="2p,darkblue")
+fig.text(x=6, y=6.5, text="fig.plot(style)", **minorexplain)
+# ============fig.plot (pen)
+fig.plot(x=4, y=6, style="c0.7c", pen="2p,darkblue")
+fig.text(x=3, y=4.5, text="fig.plot(pen)", **minorexplain)
+# ============Legend
+fig.legend()
+fig.text(x=8, y=17, text="Legend", **mainexplain)
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:80]:
```suggestion
fig.text(x=8, y=16.9, text="Legend", **mainexplain)
```

File: examples/get_started/03_figure_element.py [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+# ============Title
+fig.text(x=7.5, y=22, text='frame="+tTitle"', **minorexplain)
+# ============xlabel
+fig.text(x=5, y=-3, text='frame="x+lxlabel"', **minorexplain)
+# ============ylabel
+fig.text(x=-1.7, y=10, text='frame="y+lylabel"', angle=90, **minorexplain)
+# ============x-majorticks
+fig.plot(x=10, y=-0.2, style="c1c", pen="2p,darkblue", no_clip=True)
+fig.text(x=10, y=-1.4, text="Annotation", **mainexplain)
+fig.text(x=10, y=-2.6, text='frame="xa2"', **minorexplain)
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:52]:
```suggestion
fig.text(x=10, y=-1.6, text="Annotation", **mainexplain)
fig.text(x=10, y=-2.8, text='frame="xa2"', **minorexplain)
```

File: examples/get_started/03_figure_element.py [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+# ============y-majorticks
+fig.plot(x=-0.2, y=20, style="c1c", pen="2p,darkblue", no_clip=True)
+fig.text(x=0, y=23.5, text="Annotation", **mainexplain)
+fig.text(x=0, y=22, text='frame="ya5"', **minorexplain)
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:56]:
```suggestion
fig.text(x=0, y=23.4, text="Annotation", **mainexplain)
fig.text(x=0, y=22.2, text='frame="ya5"', **minorexplain)
```

File: examples/get_started/03_figure_element.py [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+    frame=["WStr+tTitle", "xa2f1g2+lxlabel", "ya5f1g5+lylabel"],
+)
+fig.plot(x=x, y=y_1, style="t0.3c", label="fig.plot (style)")
+fig.plot(x=x, y=y_2, pen="1.5p,red", label="fig.plot (pen)")
+
+mainexplain = {"font": "12p,2,darkblue", "justify": "TC", "no_clip": True}
+minorexplain = {"font": "10p,8", "justify": "TC", "no_clip": True}
+# ============Figure
+fig.text(x=12, y=22, text="Figure", **mainexplain)
+fig.text(x=12, y=20.5, text="pygmt.Figure()", **minorexplain)
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:42]:
```suggestion
fig.text(x=12, y=20.8, text="pygmt.Figure()", **minorexplain)
```

File: examples/get_started/03_figure_element.py [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+fig.plot(x=0, y=2, style="c0.7c", pen="2p,darkblue", no_clip=True)
+fig.text(x=-1.5, y=1, text='frame="yf1"', **minorexplain)
+# ============grid
+fig.plot(x=2, y=15, style="c0.5c", pen="2p,darkblue")
+fig.text(x=2, y=17, text="Grid", **mainexplain)
+fig.text(x=2, y=18, text='frame="xg2"', **minorexplain)
+# ============map boundaries
+fig.plot(x=10, y=9, style="c0.5c", pen="2p,darkblue", no_clip=True)
+fig.text(x=11.5, y=8, text="Map Boundary", **mainexplain)
+fig.text(x=11.5, y=6.5, text='frame="WStr"', **minorexplain)
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:71]:
```suggestion
fig.text(x=11.5, y=6.8, text='frame="WStr"', **minorexplain)
```

File: examples/get_started/03_figure_element.py [hunk 25]
```diff
@@ -0,0 +1,83 @@
...
+    region=[0, 10, 0, 20],
+    projection="X10c/8c",
+    frame=["WStr+tTitle", "xa2f1g2+lxlabel", "ya5f1g5+lylabel"],
+)
+fig.plot(x=x, y=y_1, style="t0.3c", label="fig.plot (style)")
+fig.plot(x=x, y=y_2, pen="1.5p,red", label="fig.plot (pen)")
+
+mainexplain = {"font": "12p,2,darkblue", "justify": "TC", "no_clip": True}
+minorexplain = {"font": "10p,8", "justify": "TC", "no_clip": True}
+# ============Figure
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:40]:
```suggestion
# ============ Figure
```

File: examples/get_started/03_figure_element.py [hunk 26]
```diff
@@ -0,0 +1,83 @@
...
+fig.text(x=12, y=22, text="Figure", **mainexplain)
+fig.text(x=12, y=20.5, text="pygmt.Figure()", **minorexplain)
+# ============Title
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:43]:
```suggestion
# ============ Title
```

File: examples/get_started/03_figure_element.py [hunk 27]
```diff
@@ -0,0 +1,83 @@
...
+fig.text(x=7.5, y=22, text='frame="+tTitle"', **minorexplain)
+# ============xlabel
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:45]:
```suggestion
# ============ xlabel
```

File: examples/get_started/03_figure_element.py [hunk 28]
```diff
@@ -0,0 +1,83 @@
...
+fig.text(x=5, y=-3, text='frame="x+lxlabel"', **minorexplain)
+# ============ylabel
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:47]:
```suggestion
# ============ ylabel
```

File: examples/get_started/03_figure_element.py [hunk 29]
```diff
@@ -0,0 +1,83 @@
...
+fig.text(x=-1.7, y=10, text='frame="y+lylabel"', angle=90, **minorexplain)
+# ============x-majorticks
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:49]:
```suggestion
# ============ x-majorticks
```

File: examples/get_started/03_figure_element.py [hunk 30]
```diff
@@ -0,0 +1,83 @@
...
+fig.plot(x=10, y=-0.2, style="c1c", pen="2p,darkblue", no_clip=True)
+fig.text(x=10, y=-1.4, text="Annotation", **mainexplain)
+fig.text(x=10, y=-2.6, text='frame="xa2"', **minorexplain)
+# ============y-majorticks
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:53]:
```suggestion
# ============ y-majorticks
```

File: examples/get_started/03_figure_element.py [hunk 31]
```diff
@@ -0,0 +1,83 @@
...
+fig.plot(x=-0.2, y=20, style="c1c", pen="2p,darkblue", no_clip=True)
+fig.text(x=0, y=23.5, text="Annotation", **mainexplain)
+fig.text(x=0, y=22, text='frame="ya5"', **minorexplain)
+# ============x-minorticks
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:57]:
```suggestion
# ============ x-minorticks
```

File: examples/get_started/03_figure_element.py [hunk 32]
```diff
@@ -0,0 +1,83 @@
...
+fig.plot(x=1, y=-0.2, style="c0.7c", pen="2p,darkblue", no_clip=True)
+fig.text(x=1, y=-1.4, text="Frame", **mainexplain)
+fig.text(x=1, y=-2.6, text='frame="xf1"', **minorexplain)
+# ============y-minorticks
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:61]:
```suggestion
# ============ y-minorticks
```

File: examples/get_started/03_figure_element.py [hunk 33]
```diff
@@ -0,0 +1,83 @@
...
+fig.plot(x=0, y=2, style="c0.7c", pen="2p,darkblue", no_clip=True)
+fig.text(x=-1.5, y=1, text='frame="yf1"', **minorexplain)
+# ============grid
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:64]:
```suggestion
# ============ Grid
```

File: examples/get_started/03_figure_element.py [hunk 34]
```diff
@@ -0,0 +1,83 @@
...
+fig.plot(x=2, y=15, style="c0.5c", pen="2p,darkblue")
+fig.text(x=2, y=17, text="Grid", **mainexplain)
+fig.text(x=2, y=18, text='frame="xg2"', **minorexplain)
+# ============map boundaries
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:68]:
```suggestion
# ============ Map Boundaries
```

File: examples/get_started/03_figure_element.py [hunk 35]
```diff
@@ -0,0 +1,83 @@
...
+fig.plot(x=10, y=9, style="c0.5c", pen="2p,darkblue", no_clip=True)
+fig.text(x=11.5, y=8, text="Map Boundary", **mainexplain)
+fig.text(x=11.5, y=6.5, text='frame="WStr"', **minorexplain)
+# ============fig.plot (style)
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:72]:
```suggestion
# ============ fig.plot (style)
```

File: examples/get_started/03_figure_element.py [hunk 36]
```diff
@@ -0,0 +1,83 @@
...
+fig.plot(x=6, y=8, style="c0.7c", pen="2p,darkblue")
+fig.text(x=6, y=6.5, text="fig.plot(style)", **minorexplain)
+# ============fig.plot (pen)
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:75]:
```suggestion
# ============ fig.plot (pen)
```

File: examples/get_started/03_figure_element.py [hunk 37]
```diff
@@ -0,0 +1,83 @@
...
+fig.plot(x=4, y=6, style="c0.7c", pen="2p,darkblue")
+fig.text(x=3, y=4.5, text="fig.plot(pen)", **minorexplain)
+# ============Legend
```
Comment (michaelgrund) [inline examples/get_started/03_figure_element.py:78]:
```suggestion
# ============ Legend
```

**Review by michaelgrund — COMMENTED** [review]
Some small adjustments to have an almost consistent style regarding the distances between different text boxes.

File: examples/get_started/03_figure_element.py [hunk 38]
```diff
@@ -0,0 +1,83 @@
+"""
+3. Figure elements
+==================
+The figure below shows the naming of figure elements in PyGMT.
+
+- :meth:`pygmt.Figure()`: having a number of plotting methods. Every new
+  figure must start with the creation of a :meth:`pygmt.Figure()` instance
+- ``frame``: setting plot or map boundaries (a combination of the single
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:8]: Just remove the trailing white spaces to make the Code Style checks happy 🙂.
```suggestion
- ``frame``: setting plot or map boundaries (a combination of the single
```

File: examples/get_started/03_figure_element.py [hunk 39]
```diff
@@ -0,0 +1,83 @@
+"""
+3. Figure elements
+==================
+The figure below shows the naming of figure elements in PyGMT.
+
+- :meth:`pygmt.Figure()`: having a number of plotting methods. Every new
+  figure must start with the creation of a :meth:`pygmt.Figure()` instance
+- ``frame``: setting plot or map boundaries (a combination of the single 
+  letters of **WSNE**, **wsne**, or **lbtr**), adding annotations, ticks,
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:9]:
```suggestion
  letters of **WSNE**, **wsne**, or **lbtr**), adding annotations, ticks,
```

File: examples/get_started/03_figure_element.py [hunk 40]
```diff
@@ -0,0 +1,83 @@
+"""
+3. Figure elements
+==================
+The figure below shows the naming of figure elements in PyGMT.
+
+- :meth:`pygmt.Figure()`: having a number of plotting methods. Every new
+  figure must start with the creation of a :meth:`pygmt.Figure()` instance
+- ``frame``: setting plot or map boundaries (a combination of the single 
+  letters of **WSNE**, **wsne**, or **lbtr**), adding annotations, ticks, 
+  gridlines (**afg**), axis labels (**+l**), and title (**+t**), e.g.,
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:10]:
```suggestion
  gridlines (**afg**), axis labels (**+l**), and title (**+t**), e.g.,
```

File: examples/get_started/03_figure_element.py [hunk 41]
```diff
@@ -0,0 +1,83 @@
...
+3. Figure elements
+==================
+The figure below shows the naming of figure elements in PyGMT.
+
+- :meth:`pygmt.Figure()`: having a number of plotting methods. Every new
+  figure must start with the creation of a :meth:`pygmt.Figure()` instance
+- ``frame``: setting plot or map boundaries (a combination of the single 
+  letters of **WSNE**, **wsne**, or **lbtr**), adding annotations, ticks, 
+  gridlines (**afg**), axis labels (**+l**), and title (**+t**), e.g., 
+  in :meth:`pygmt.Figure.basemap`. Detailed examples can be found at
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:11]:
```suggestion
  in :meth:`pygmt.Figure.basemap`. Detailed examples can be found at
```

File: examples/get_started/03_figure_element.py [hunk 42]
```diff
@@ -0,0 +1,83 @@
...
+  gridlines (**afg**), axis labels (**+l**), and title (**+t**), e.g., 
+  in :meth:`pygmt.Figure.basemap`. Detailed examples can be found at 
+  :doc:`frame and axes attributes </tutorials/basics/frames>`
+- :meth:`pygmt.Figure.plot`: plotting lines or symbols based on ``pen``
+  or ``style`` parameters, respectively
+- :meth:`pygmt.Figure.text`: plotting text strings whereby the ``font``
+  parameter adjusts fontsize, fontstyle, and color
+- :meth:`pygmt.Figure.legend`: showing the naming of lines or symbols while
+  the ``label`` is given in :meth:`pygmt.Figure.plot`
+- :meth:`pygmt.Figure.show`: previewing the content added to the current
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:19]:
```suggestion
- :meth:`pygmt.Figure.show`: previewing the content added to the current
```

**Review by michaelgrund — APPROVED** [review]

File: examples/get_started/03_figure_element.py [hunk 43]
```diff
@@ -0,0 +1,83 @@
...
+fig.plot(x=1, y=-0.2, style="c0.7c", pen="2p,darkblue", no_clip=True)
+fig.text(x=1, y=-1.4, text="Frame", **mainexplain)
+fig.text(x=1, y=-2.6, text='frame="xf1"', **minorexplain)
+# ============ y-minorticks
+fig.plot(x=0, y=2, style="c0.7c", pen="2p,darkblue", no_clip=True)
+fig.text(x=-1.5, y=1, text='frame="yf1"', **minorexplain)
+# ============ Grid
+fig.plot(x=2, y=15, style="c0.5c", pen="2p,darkblue")
+fig.text(x=2, y=17, text="Grid", **mainexplain)
+fig.text(x=2, y=18, text='frame="xg2"', **minorexplain)
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:67]: At the marked position (a crossing point of x and y gridlines) it is strictly speaking:
```suggestion
fig.text(x=2, y=18, text='frame="[xg2, yg5]"', **minorexplain)
```

File: examples/get_started/03_figure_element.py [hunk 44]
```diff
@@ -0,0 +1,83 @@
...
+# ============ Map Boundaries
+fig.plot(x=10, y=9, style="c0.5c", pen="2p,darkblue", no_clip=True)
+fig.text(x=11.5, y=8, text="Map Boundary", **mainexplain)
+fig.text(x=11.5, y=6.8, text='frame="WStr"', **minorexplain)
+# ============ fig.plot (style)
+fig.plot(x=6, y=8, style="c0.7c", pen="2p,darkblue")
+fig.text(x=6, y=6.5, text="fig.plot(style)", **minorexplain)
+# ============ fig.plot (pen)
+fig.plot(x=4, y=6, style="c0.7c", pen="2p,darkblue")
+fig.text(x=3, y=4.5, text="fig.plot(pen)", **minorexplain)
```
Comment (yvonnefroehlich) [inline examples/get_started/03_figure_element.py:77]: Small adjustment of the position of the "fig.plot(style)" label:
```suggestion
# ============ fig.plot (style)
fig.plot(x=6, y=8, style="c0.7c", pen="2p,darkblue")
fig.text(x=7, y=6.5, text="fig.plot(style)", **minorexplain)
# ============ fig.plot (pen)
fig.plot(x=4, y=6, style="c0.7c", pen="2p,darkblue")
fig.text(x=3, y=4.5, text="fig.plot(pen)", **minorexplain)
```
| currently  | suggestion |
| -------- | ------- |
|fig_plot_style_label_position_old | fig_plot_style_label_position_new |

**Review by yvonnefroehlich — COMMENTED** [review]
Thanks @JingHuiTong for working on this tutorial :slightly_smiling_face:. It will definitely help users coming from other python plotting libraries or MATLAB.

**Comment by weiji14** [comment]
I'm gonna go ahead and apply Yvonne's suggestions and merge this PR for PyGMT v0.10.0.

**Review by seisman — APPROVED** [review]

**Pull request merged**